pkg/controllers/controlplanemachinesetgenerator/controller_test.go

```diff
@@ -1870,3 +1871,719 @@ var _ = Describe("controlplanemachinesetgenerator controller on GCP", func() {
+						Eventually(komega.Get(cpms)).Should(Succeed())
+
+						Expect(cpms.Spec.Template.OpenShiftMachineV1Beta1Machine.FailureDomains).To(Equal(cpms5FailureDomainsBuilderOpenStack.BuildFailureDomains()))
+					})
+				})
+			})
+		})
+
+		Context("with 3 Machine Sets", func() {
+			BeforeEach(func() {
+				By("Creating MachineSets")
+				create3MachineSets()
+			})
+
+			Context("with 3 existing control plane machines", func() {
+				BeforeEach(func() {
+					By("Creating Control Plane Machines")
+					create3CPMachines()
+				})
+
+				It("should create the ControlPlaneMachineSet with the expected fields", func() {
+					By("Checking the Control Plane Machine Set has been created")
+					Eventually(komega.Get(cpms)).Should(Succeed())
+					Expect(cpms.Spec.State).To(Equal(machinev1.ControlPlaneMachineSetStateInactive))
+					Expect(*cpms.Spec.Replicas).To(Equal(int32(3)))
+				})
+
+				It("should create the ControlPlaneMachineSet with the provider spec matching the youngest machine provider spec", func() {
+					By("Checking the Control Plane Machine Set has been created")
+					Eventually(komega.Get(cpms)).Should(Succeed())
+					// In this case expect the machine Provider Spec of the youngest machine to be used here.
+					// In this case it should be `machine-2` given that's the one we created last.
+					cpmsProviderSpec, err := providerconfig.NewProviderConfigFromMachineSpec(cpms.Spec.Template.OpenShiftMachineV1Beta1Machine.Spec)
+					Expect(err).To(BeNil())
+
+					machineProviderSpec, err := providerconfig.NewProviderConfigFromMachineSpec(machine2.Spec)
+					Expect(err).To(BeNil())
+
+					// Remove from the machine Provider Spec the fields that won't be
+					// present on the ControlPlaneMachineSet Provider Spec.
+					openStackMachineProviderConfig := machineProviderSpec.OpenStack().Config()
+					if openStackMachineProviderConfig.AvailabilityZone != "" {
+						openStackMachineProviderConfig.AvailabilityZone = ""
+					}
+					if openStackMachineProviderConfig.RootVolume != nil && openStackMachineProviderConfig.RootVolume.Zone != "" {
+						openStackMachineProviderConfig.RootVolume.Zone = ""
+					}
+
+					Expect(cpmsProviderSpec.OpenStack().Config()).To(Equal(openStackMachineProviderConfig))
+				})
+
+				It("should create the ControlPlaneMachineSet with only one copy of each of the 3 failure domains", func() {
+					By("Checking the Control Plane Machine Set has been created")
+					Eventually(komega.Get(cpms)).Should(Succeed())
+
+					Expect(cpms.Spec.Template.OpenShiftMachineV1Beta1Machine.FailureDomains).To(Equal(cpms3FailureDomainsBuilderOpenStack.BuildFailureDomains()))
+				})
+
+				Context("With additional Machines adding additional failure domains", func() {
+					BeforeEach(func() {
+						By("Creating additional Machines")
+						createAZ4Machine()
+						createAZ5Machine()
+					})
+
+					FIt("should create the ControlPlaneMachineSet with only one copy of each the 5 failure domains", func() {
```

Comment:
It means there is a focus set on one of the tests, 
```suggestion
					It("should create the ControlPlaneMachineSet with only one copy of each the 5 failure domains", func() {
```